File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    int len=arr.size();
    
    int st=arr.at(0);
    answer.push_back(st);
    for(int i=1;i<len;i++){
        int cur=arr.at(i);
        if(st!=cur){
            answer.push_back(cur);
        }
        st=cur;
    }

    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        
        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
        System.out.println("Hello Java");
        
        Deque<Integer> dq=new ArrayDeque<>();
        
        for(int x:arr){
            if(dq.isEmpty()){
                dq.addLast(x);
            }
            else{
                int l=dq.peekLast();
                if(l!=x){
                    dq.addLast(x);
                }
            }
        }
        
        int dql=dq.size();
        answer=new int[dql];
        for(int i=0;i<dql;i++){
            answer[i]=dq.pollFirst();
        }
        
        return answer;
    }
}